subsys/net/lib/mqtt_sn/mqtt_sn.c

```diff
@@ -1719,3 +1738,36 @@ int mqtt_sn_get_topic_name(struct mqtt_sn_client *client, uint16_t id,
 	}
 	return -ENOENT;
 }
+
+int mqtt_sn_predefine_topic(struct mqtt_sn_client *client, uint16_t topic_id,
+			    struct mqtt_sn_data *topic_name)
+{
+	struct mqtt_sn_topic *topic;
+
+	if (!client || !topic_name) {
```

Comment:
Please change all these here and all over your changes to use explicit checks for non boolean variables as there is a misra rule telling that. We currently have no compliance checks for these but it is probably coming at some point.
So here `if (client == NULL || topic_name == NULL)` etc

Reply:
done